docs/guide.rst

```diff
@@ -101,16 +101,64 @@ high-level :meth:`Contract.from_address` method to get a contract instance.
 :meth:`Contract.from_address` works with contracts which are not proxies, so it is the most universal method of getting
 a contract not knowing the abi.
 
-Check out the code!
+ProxyChecks
+###########
+
+Since the Proxy contracts on StarkNet can have different implementations (especially, since everyone can write their own), there is no single way of checking if some contract is a Proxy contract.
+
+Also, there are two ways of proxying a contract on StarkNet:
+ - make `implementation` in a Proxy contract store a `class hash` of the proxied contract
+ - make `implementation` in a Proxy contract store an `address` of the proxied contract
```

Comment:
I think this is not 100% correct. Afaik all that is required is to have a `__default__` entrypoint that does some forwarding logic. Where exactly is the implementation stored is not defined I think.